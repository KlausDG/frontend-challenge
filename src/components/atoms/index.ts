export { default as Typography } from './Typography/Typography.vue'
export { default as Stack } from './Stack/Stack.vue'
export { default as Card } from './Card/Card.vue'
export { default as Badge } from './Badge/Badge.vue'
