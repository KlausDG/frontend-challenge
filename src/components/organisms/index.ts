export { default as SupplierSection } from './SupplierSection/SupplierSection.vue'
export { default as ShipToSection } from './ShipToSection/ShipToSection.vue'
export { default as BillToSection } from './BillToSection/BillToSection.vue'
export { default as ChargeToSection } from './ChargeToSection/ChargeToSection.vue'
export { default as Header } from './Header/Header.vue'
