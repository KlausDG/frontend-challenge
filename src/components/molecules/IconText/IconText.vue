<script setup lang="ts">
import { Typography } from '@/components/atoms'

defineProps<{
  icon: string
  text: string
}>()
</script>

<template>
  <div class="icon-text__container">
    <unicon :name="icon" />
    <Typography>{{ text }}</Typography>
  </div>
</template>

<style scoped lang="scss">
.icon-text__container {
  display: flex;
  align-items: center;
  gap: $spacing-sm;
}
</style>
