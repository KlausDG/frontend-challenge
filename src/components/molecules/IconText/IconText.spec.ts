import { mount } from '@vue/test-utils'
import { describe, it, expect, vi } from 'vitest'
import IconText from './IconText.vue'
import { Typography } from '@/components/atoms'

vi.mock('unicon-vue', () => ({
  default: {
    name: 'Unicon',
    template: '<div class="mock-unicon"></div>',
  },
}))

describe('IconText.vue', () => {
  it('renders the Typography component with the correct text', () => {
    const wrapper = mount(IconText, {
      props: {
        icon: 'user',
        text: 'Hello IconText',
      },
    })

    const typography = wrapper.findComponent(Typography)
    expect(typography.exists()).toBe(true)
    expect(typography.text()).toBe('Hello IconText')
  })

  it('renders the unicon component with the correct icon name', () => {
    const wrapper = mount(IconText, {
      props: { icon: 'user', text: 'Hello IconText' },
      global: {
        components: {
          unicon: {
            props: ['name'],
            template: '<div class="mock-unicon">{{ name }}</div>',
          },
          Typography,
        },
      },
    })

    const unicon = wrapper.find('.mock-unicon')
    expect(unicon.exists()).toBe(true)
    expect(unicon.text()).toBe('user')
  })

  it('applies the correct container class', () => {
    const wrapper = mount(IconText, {
      props: { icon: 'phone', text: 'Phone Icon' },
    })
    expect(wrapper.classes()).toContain('icon-text__container')
  })

  it('renders multiple instances correctly', () => {
    const wrapper = mount({
      template: `
        <div>
          <IconText icon="user" text="User" />
          <IconText icon="phone" text="Phone" />
        </div>
      `,
      components: { IconText },
    })
    const instances = wrapper.findAllComponents(IconText)
    expect(instances).toHaveLength(2)
    expect(instances[0].text()).toContain('User')
    expect(instances[1].text()).toContain('Phone')
  })

  it('matches the component snapshot', () => {
    const wrapper = mount(IconText, {
      props: { icon: 'print', text: 'Print Icon' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
