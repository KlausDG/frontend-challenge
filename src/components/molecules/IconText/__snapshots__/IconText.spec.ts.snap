// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconText.vue > matches the component snapshot 1`] = `
"<div data-v-70c01fa4="" class="icon-text__container">
  <unicon data-v-70c01fa4="" name="print"></unicon>
  <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">Print Icon</p>
</div>"
`;
